apps/designer/app/designer/features/sidebar-left/panels/components/uploading-animation.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react";
+import { useInterval } from "react-use";
+
+const useFakeProgress = ({
+  isUploading,
+  isDeleting,
+}: {
+  isUploading: boolean;
+  isDeleting: boolean;
+}) => {
+  const [progressBarPercentage, setProgressBarPercentage] = useState(0);
+
+  // @todo rewrite this fake indication to show real progress
+  useInterval(
+    () => {
+      setProgressBarPercentage((percentage) =>
+        percentage < 100 ? percentage + 1 : percentage
+      );
+    },
+    isUploading || isDeleting ? 100 : null
+  );
+
+  return progressBarPercentage;
+};
+
+export const UploadingAnimation = ({
+  isUploading,
+  isDeleting,
+}: {
+  isUploading: boolean;
+  isDeleting: boolean;
+}) => {
+  const progressBarPercentage = useFakeProgress({ isUploading, isDeleting });
+  return (
+    <svg
+      style={{ position: "relative", zIndex: 10 }}
```

Comment:
lol, why not 999?

Reply:
I am an adult, i use zIndex carefully ahaha

Reply:
ok it seems youcan use Radix's Progress and pass asChild prop and render the SVG inside, something like

```
<ProgressIndicator asChild>
   <CircularProgress />
</ProgressIndicator>
```
Circular Progress will receive all the props from radix

Reply:
> I am an adult, i use zIndex carefully ahaha

we have a few predefined levels in the design system, you should be ok with the first one